datalad_next/commands/status.py

```diff
@@ -0,0 +1,200 @@
+    modified = 'modified'
+    deleted = 'deleted'
+    untracked = 'untracked'
+    unknown = 'unknown'
+
+
+STATE_COLOR_MAP = {
+    StatusState.added: ac.GREEN,
+    StatusState.modified: ac.RED,
+    StatusState.deleted: ac.RED,
+    StatusState.untracked: ac.RED,
+    StatusState.unknown: ac.YELLOW,
+}
+
+
+diffstatus2resultstate_map = {
+    GitDiffStatus.addition: StatusState.added,
+    GitDiffStatus.copy: StatusState.added,
+    GitDiffStatus.deletion: StatusState.deleted,
+    GitDiffStatus.modification: StatusState.modified,
+    GitDiffStatus.rename: StatusState.added,
+    GitDiffStatus.typechange: StatusState.modified,
+    GitDiffStatus.unmerged: StatusState.unknown,
+    GitDiffStatus.unknown: StatusState.unknown,
+    GitDiffStatus.other: StatusState.untracked,
+}
+
+
+@dataclass(kw_only=True)
+class StatusResult(CommandResult):
+    # named ...state to emphasize conceptual similarity
+    # with legacy property 'state'
+    diff_state: GitDiffStatus
+    type: GitTreeItemType
+    prev_type: GitTreeItemType
+
+    @property
+    def state(self) -> StatusState:
+        """A (more or less legacy) simplified representation of the subject
+        state. For a more accurate classification use the ``diff_status``
+        property.
+        """
+        return diffstatus2resultstate_map[self.diff_state]
+
+    @property
+    def type_src(self) -> str | None:
+        """Backward-compatibility adaptor"""
+        return self.prev_type
+
+
+@build_doc
+class Status(ValidatedInterface):
+    """The is a previous of an upcoming command implementation to replace
+    the DataLad ``status`` command.
+
+    For now expect anything here to change again.
+    """
+    # Interface.validate_args() will inspect this dict for the presence of a
+    # validator for particular parameters
+    _validator_ = EnsureCommandParameterization(
+        param_constraints=dict(
+            # if given, it must also exist
+            dataset=EnsureDataset(installed=True),
+            untracked=EnsureChoice(
+                'no', 'whole-dir', 'no-empty-dir', 'normal', 'all'),
+        ),
+        validate_defaults=('dataset',),
+    )
+
+    # this is largely here for documentation and CLI parser building
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""Dataset to be used as a configuration source. Beyond
+            reading configuration items, this command does not interact with
+            the dataset."""),
+        untracked=Parameter(
+            args=('--untracked',),
+            nargs='?',
+            const='no-empty-dir',
+            doc="""If and how untracked content is reported when comparing
+            a revision to the state of the working tree. 'no': no untracked
+            content is reported; 'normal': untracked files and entire
+            untracked directories are reported as such; 'all': report
+            individual files even in fully untracked directories."""),
```

Comment:
I tried this, and `no` and `normal` aren't yet implemented, and produce a plain ``[ERROR]``. I assume you know, but wanted to leave a comment nevertheless

```
(handbook-upgrade) adina@muninn in ~/repos/datalad-next on git:gitstatus
❱ datalad next-status --untracked no
[ERROR  ] 'no' 
(handbook-upgrade) adina@muninn in ~/repos/datalad-next on git:gitstatus
❱ datalad next-status --untracked normal                                    1 !
[ERROR  ] 'normal' 
(handbook-upgrade) adina@muninn in ~/repos/datalad-next on git:gitstatus
❱ datalad next-status --untracked all                                       1 !
untracked: .idea/datalad-next.iml
untracked: .idea/inspectionProfiles/profiles_settings.xml
untracked: .idea/modules.xml
untracked: .idea/vcs.xml
untracked: .idea/workspace.xml
untracked: some
(handbook-upgrade) adina@muninn in ~/repos/datalad-next on git:gitstatus
❱ datalad next-status --untracked whole-dir
untracked: .idea
untracked: .mypy_cache
untracked: .pytest_cache
untracked: __pycache__
```

Reply:
Addressed locally now. Thx!